src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -0,0 +1,98 @@
+package org.broadinstitute.dsde.workbench.sam.openam
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.effect.{ContextShift, IO}
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.workbench.sam.errorReportSource
+import org.broadinstitute.dsde.workbench.model._
+import org.broadinstitute.dsde.workbench.sam.db.DbReference
+import org.broadinstitute.dsde.workbench.sam.db.SamParameterBinderFactory._
+import org.broadinstitute.dsde.workbench.sam.db.tables._
+import org.broadinstitute.dsde.workbench.sam.model._
+import org.broadinstitute.dsde.workbench.sam.util.DatabaseSupport
+import org.postgresql.util.PSQLException
+import scalikejdbc._
+
+import scala.concurrent.ExecutionContext
+
+class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
+                              protected val ecForDatabaseIO: ExecutionContext)(implicit executionContext: ExecutionContext) extends AccessPolicyDAO with DatabaseSupport with LazyLogging {
+
+  implicit val cs: ContextShift[IO] = IO.contextShift(executionContext)
+
+  def createResourceType(resourceTypeName: ResourceTypeName): IO[ResourceTypeName] = {
+    runInTransaction { implicit session =>
+      insertResourceType(resourceTypeName)
+      resourceTypeName
+    }
+  }
+
+  private def insertResourceType(resourceTypeName: ResourceTypeName)(implicit session: DBSession): ResourceTypePK = {
+    val resourceTypeTableColumn = ResourceTypeTable.column
+    val insertResourceTypeQuery = samsql"""insert into ${ResourceTypeTable.table} (${resourceTypeTableColumn.name}) values (${resourceTypeName.value})"""
+
+    ResourceTypePK(insertResourceTypeQuery.updateAndReturnGeneratedKey().apply())
+  }
+
+  // 1. Create Resource
+  // 2. Create the entries in the join table for the auth domains
+  def createResource(resource: Resource): IO[Resource] = {
+    runInTransaction { implicit session =>
+      val resourcePK = insertResource(resource)
+
+      if (resource.authDomain.nonEmpty) {
+        insertAuthDomainsForResource(resourcePK, resource.authDomain)
+      }
+
+      resource
+    }.recoverWith {
+      case sqlException: PSQLException => {
+        logger.debug(s"createResource psql exception on resource $resource", sqlException)
+        IO.raiseError(new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.Conflict, s"Create resource failed with the following database exception: ${sqlException.getMessage}")))
```

Comment:
`ErrorReport` can take an exception in its constructor, use that instead of embedding the exception message.